core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/WorkflowPipelinedRegions.scala

```diff
@@ -0,0 +1,166 @@
+package edu.uci.ics.amber.engine.architecture.scheduling
+
+import edu.uci.ics.amber.engine.architecture.controller.Workflow
+import edu.uci.ics.amber.engine.common.virtualidentity.util.toOperatorIdentity
+import edu.uci.ics.amber.engine.common.virtualidentity.{LinkIdentity, OperatorIdentity}
+import edu.uci.ics.texera.workflow.common.workflow.OperatorLink
+import org.jgrapht.alg.connectivity.BiconnectivityInspector
+import org.jgrapht.graph.DirectedAcyclicGraph
+
+import scala.collection.immutable.HashMap
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+
+case class PipelinedRegionDAGLink(
+    originRegionId: PipelinedRegionIdentity,
+    destinationRegion: PipelinedRegionIdentity
+)
+
+class WorkflowPipelinedRegions(workflow: Workflow) {
+  var pipelinedRegionsDAG: DirectedAcyclicGraph[PipelinedRegion, PipelinedRegionDAGLink] = null
```

Comment:
you can use DefaultEdge
https://jgrapht.org/javadoc/org.jgrapht.core/org/jgrapht/graph/DefaultEdge.html

Reply:
done